filebeat/input/filestream/input.go

```diff
@@ -85,28 +86,66 @@ func Plugin(log *logp.Logger, store statestore.States) input.Plugin {
 	}
 }
 
+func updateDefaults(cfg config, inactiveSet bool) config {
+	defaultInactive := defaultCloserConfig().OnStateChange.Inactive
+	if cfg.Delete.OnClose.Inactive && !inactiveSet {
+		// This means the `close.on_state_change.inactive` is either not set or
+		// it is set to the default value, in either case, overwrite it to
+		// 30min
+		if cfg.Close.OnStateChange.Inactive == defaultInactive {
```

Comment:
What's the reason for this extra default check? `inactiveSet` is false, so we already know there was no setting for this flag, and the config was unpacked onto default settings, so isn't this redundant?

Reply:
It's an extra caution, not to rely on the behaviour of the caller. But you're right, it is redundant here.